### PR TITLE
enhance worker run loop

### PR DESCRIPTION
**Pull request by hexaclock**

### CODE REVIEW

File: cheroot/workers/threadpool.py [hunk 1]
```diff
@@ -7,6 +7,7 @@
 import threading
 import time
 import socket
+import select
```
Comment (webknjaz) [inline cheroot/workers/threadpool.py:10]: How about using epoll? https://github.com/cherrypy/cherrypy/issues/769#issuecomment-274639308

File: cheroot/workers/threadpool.py [hunk 2]
```diff
@@ -100,25 +101,46 @@ def run(self):
...
+                else:
+                    if conn is _SHUTDOWNREQUEST:
+                        return
+                    conn_socks[conn.socket] = (conn, time.time())
+
+                rlist, wlist, xlist = [], [], []
+                socks = [sock for sock in conn_socks.keys() if sock.fileno() > -1]
+                if socks:
+                    rlist, wlist, xlist = select.select(socks, [], [], 0)
+                for sock in rlist + wlist + xlist:
```
Comment (webknjaz) [inline cheroot/workers/threadpool.py]: I'd use `itertools.chain` here

File: cheroot/workers/threadpool.py (hunk 1 above)
Comment (hexaclock) [reply cheroot/workers/threadpool.py:10]: epoll is only supported on Linux

Comment (webknjaz) [reply cheroot/workers/threadpool.py:10]: Yeah, so it'd need fallbacks

**Pull request merged**